composer/models/huggingface.py

```diff
@@ -74,6 +94,140 @@ def __init__(self,
+            from composer.trainer import Trainer
+
+            hf_model = transformers.AutoModelForSequenceClassification.from_pretrained('prajjwal1/bert-tiny', num_labels=2)
+            hf_tokenizer = transformers.AutoTokenizer.from_pretrained('prajjwal1/bert-tiny')
+            composer_model = HuggingFaceModel(hf_model, tokenizer=hf_tokenizer, metrics=[], use_logits=True)
+            trainer = Trainer(model=composer_model,
+                              train_dataloader=train_dataloader,
+                              save_filename='composer-hf-checkpoint.pt',
+                              max_duration='1ep',
+                              save_folder='./')
+            trainer.fit()
+            trainer.close()
+
+        Example:
+
+        .. testcode::
+
+            hf_model, hf_tokenizer = HuggingFaceModel.hf_from_composer_checkpoint('composer-hf-checkpoint.pt')
+            composer_model = HuggingFaceModel(hf_model, hf_tokenizer)
+            trainer = Trainer(model=composer_model,
+                              train_dataloader=train_dataloader,
+                              save_filename='composer-hf-checkpoint-2.pt',
+                              max_duration='1ep',
+                              save_folder='./')
+
+        Args:
+            checkpoint_path (str): Path to the composer checkpoint, can be a local path, ``http(s)://`` url, or ``s3://`` uri
+            model_instantiation_class (Union[Type[:class:`transformers.PreTrainedModel`], Type[:class:`transformers.AutoModel`]]), optional):
+                Class to use to create the HuggingFace model. Defaults to the model class used in the original checkpoint. If this argument is
+                a HuggingFace auto class (e.g. :class:`transformers.AutoModel` or :class:`transformers.AutoModelForSequenceClassification`), the ``from_config`` method will be used,
+                while if it is of type :class:`transformers.PreTrainedModel`, the constructor will be called.
+            model_init_kwargs: Dict[str, Any]: Extra arguments to pass in for the model creation (e.g. ``num_labels`` for creating a sequence classification model)
+            local_checkpoint_save_location (Optional[Union[Path, str]], optional): If specified, where to save the checkpoint file to locally.
+                                                                                   If the input ``checkpoint_path`` is already a local path, this will be a symlink.
+                                                                                   Defaults to None, which will use a temporary file.
+
+        Raises:
+            ValueError: If the ``model_instantiation_class``, or the model class saved in the checkpoint, is not able to be imported
+
+        Returns:
+            Tuple[transformers.PreTrainedModel, Optional[transformers.PreTrainedTokenizer]]: The loaded HuggingFace model and (if present) tokenizer
+        """
+        import transformers
```

Comment:
wrap in MissingConditionalImportError?

Reply:
It seemed like we only did that in class inits, not everywhere the import is used. But i guess you can use this function without calling the class init? I can wrap this if you think that makes sense.

Reply:
wrapped